src/Model/Changelog/Changelog.php

```diff
@@ -117,6 +117,56 @@ protected function getChanges(OutputInterface $output)
         return $this->sortByDate($changes);
     }
 
+    /**
+     * Returns the data for rendering the changelogs.
+     */
+    public function getChangesRenderData(OutputInterface $output): array
+    {
+        $libraries = $this->getRootLibrary()->getAllItems(true);
+
+        $libs = [];
+        $logs = [];
+        $commitToLibraryMap = [];
+
+        foreach ($libraries as $library) {
+            $libLogs = $this->sortByDate($this->getLibraryLog($output, $library));
+            $composerData = $library->getRelease()->getLibrary()->getComposerData();
+            $name = strtolower(trim($composerData['name']));
+
+            $libs[$name] = [
+                'name' => $name,
+                'link' => 'https://packagist.org/packages/' . $name,
+                'version' => [
+                    'prior' => $library->getRelease()->getPriorVersion()->getValue(),
+                    'release' => $library->getRelease()->getVersion()->getValue()
+                ],
+                'commits' => [
+                    'all' => $libLogs,
+                    'by_type' => $this->sortByType($libLogs)
+                ]
+            ];
+
+            $logs = array_merge($logs, $libLogs);
+
+            foreach ($libLogs as $log) {
+                $commitToLibraryMap[$log->getShortHash()] = $name;
+            }
+        }
+
+        $logs = $this->sortByDate($logs);
+
+        $data = [
+            'libs' => $libs,
+            'commits' => [
+                'all' => $logs,
+                'by_type' => $this->sortByType($logs),
+                'map_to_lib' => $commitToLibraryMap
+            ]
+        ];
+
+        return array_map($fetchChangeData, $data);
```

Comment:
@dnsl48 Running changelog generation locally pointed out to me that `$fetchChangeData` here is never set - where is this function intended to come from?

Reply:
Yes, good catch. That's left from the original version that hydrated the data array on the backend. However, that's been too bad for performance and memory consumption, so we're doing that in templates now. Forgot to remove the map.

Reply:
fixed